tools/src/main/scala/scala/scalanative/linker/Reach.scala

```diff
@@ -76,14 +83,15 @@ class Reach(config: build.Config, entries: Seq[Global], loader: ClassLoader) {
 
   def process(): Unit =
     while (todo.nonEmpty) {
-      val name = todo.pop()
+      val name = todo.last
+      todo = todo.init
       if (!done.contains(name)) {
         reachDefn(name)
       }
     }
 
   def reachDefn(name: Global): Unit = {
-    stack.push(name)
+    stack :+= name
```

Comment:
Have you considered to use a `mutable.ListBuffer` instead of a `var` of a `List`?
Appending to a `List` like this has a linear cost and creates garbage.

Reply:
I did not like that either, but the deprecation note says:

> Stack is an inelegant and potentially poorly-performing wrapper around List. **Use a List assigned to a var instead.**

Reply:
In that case you need to push on the left (using `stack ::= m`) instead of pushing on the right, so you don't recreate an entire new List from scratch on every push.

Reply:
Oh, by _that_ you meant using a `var` of a `List`? Now I understand your answer..
Well, I think that If you push on the head (as I said in the last comment) you get a pretty performant stack, but I wouldn't bet anything on it being better/worse than `mutable.ListBuffer` 😄